.travis.yml

```diff
@@ -1,42 +1,51 @@
 sudo: false
 language: python
-services:
-    - mongodb
 
-matrix:
-  fast_finish: true
+jobs:
   include:
-    - env: TOXENV=flake8
+    - stage: pretest
+      env: TOXENV=flake8
       python: 3.6
     - env: TOXENV=pylint
       python: 3.6
     - env: TOXENV=doc8
       python: 3.6
-    - env: TOXENV=packaging
-      python: 3.6
     - env: TOXENV=docs
       python: 3.6
-    - env: TOXENV=py35
+    - &test
+      stage: test
+      services: mongodb
+      before_script:
+        - sleep 15
+        - mongo orion_test --eval 'db.createUser({user:"user",pwd:"pass",roles:["readWrite"]});'
+      after_success:
+        - tox -e final-coverage
+        - tox -e codecov
+      env: TOXENV=py35
       python: 3.5
-    - env: TOXENV=py36
+    - <<: *test
+      env: TOXENV=py36
       python: 3.6
-    - env: TOXENV=py37
+    - <<: *test
+      env: TOXENV=py37
       python: 3.7
       dist: xenial
       sudo: true
-    - env: TOXENV=demo_random
+    - <<: *test
+      env: TOXENV=demo_random
+      python: 3.6
+    - stage: packaging
+      env: TOXENV=packaging
```

Comment:
The naming of toxenv is probably bad here, `packaging` is supposed to be a pretest.

Reply:
The following test is a deployment to conda, I guess he wanted to bundle packaging and conda upload together in post-test stage. @abergeron How is the deployment working? Shouldn't it only upload when tested with the master branch?

Reply:
I always thought that packaging was to be done last, but if you want it as a pretest, we can move it.

As for the conda packages, since packaging and (conditional) upload are bundled it might be better to put it last.

Reply:
What is the condition for the upload? If we are to keep conda packaging and upload bundled I would keep the other packaging test at this stage for coherence.

Reply:
It will upload the package if it is building a tag.

Reply:
Oh ok, just checked the docs about [TRAVIS_TAG](https://github.com/Epistimio/orion/pull/141/files#diff-00041ed7660060aca02d68ff8d10fbd7R17)? I didn't know this was how it gets set. I remember you telling me about testing the build with a tag.

Reply:
This builds the package on every run, but only uploads for tags.

So what I meant is that it would be good once this is merge to add some tag in order to test that the upload process works correctly.

Reply:
We don't need to wait for a merge. I made a new branch in the repo and tagged it to test the upload. 

Branch:
https://github.com/Epistimio/orion/tree/feature/conda

Tag:
https://github.com/Epistimio/orion/releases/tag/test-conda-upload

Travis test:
https://travis-ci.org/Epistimio/orion/builds/442688743

Reply:
@abergeron It seems anaconda is not installed properly.

> ./conda/conda_build.sh: line 20: anaconda: command not found

https://travis-ci.org/Epistimio/orion/jobs/442688753#L2681

Reply:
Yep, anaconda-client was missing, I fixed that.

Reply:
You can force push a new commit id for the test-conda-upload tag to make sure it works, or create a new one.

Reply:
I made a new tag with commit cfbe4ee:
https://github.com/Epistimio/orion/releases/tag/test-conda-upload-1.1

Travis test:
https://travis-ci.org/Epistimio/orion/builds/442800952

Reply:
It worked and the packages are on ananconda.org: 

https://anaconda.org/epistimio/orion

Although you might want to delete those since they essentially from a random commit and not a real release.

Reply:
Yes thanks. I was waiting to install conda and test that the available packages are fine. On my Ubuntu system at least.

Reply:
There seems to be a problem with the packages on conda. I can't find it.

```bash
$ conda install -c epistimio orion
Fetching package metadata .............

PackageNotFoundError: Packages missing in current channels:
            
  - orion
```

```bash
$ conda search -c epistimio --override-channels
Loading channels: done

PackagesNotFoundError: The following packages are not available from current channels:

  - *
```

Could it be because package name is specified as `orion` rather than `orion.core` in the conda [config](https://github.com/Epistimio/orion/pull/141/files#diff-dbb380fc11ea56e32b4687905343af66R4)?

Reply:
It works for me.

Reply:
Oh I get it somehow, there aren't any packages for linux-64.  This is only config missing, which is probably because the build packages are for that and it skips the conversion, which makes us skip the upload.

I'll have to fix that.

Reply:
I updated tag test-conda-upload

Travis test:
https://travis-ci.org/Epistimio/orion/builds/444765160